pkg/bb/bbmain/register.go

```diff
@@ -18,6 +19,60 @@ var ErrNotRegistered = errors.New("command not registered")
 // Noop is a noop function.
 var Noop = func() {}
 
+// ListCmds lists bb commands and verifies symlinks.
+// It is by convention called when the bb command is invoked directly.
+// For every command, there should be a symlink in /bbin,
+// and for every symlink, there should be a command.
+// Occasionally, we have bugs that result in one of these
+// being false. Just running bb is an easy way to tell if something
+// in your image is messed up.
+func ListCmds() {
+	type known struct {
+		name string
+		bb   string
+	}
+	names := map[string]*known{}
+	g, err := filepath.Glob("/bbin/*")
```

Comment:
No. Why do all this bbin work? Just list the commands in bbCmds. Nothing more, nothing less.

The user can always invoke them using ./bb <cmd> instead of symlinks...

If you do want the symlinks, make it an optional feature with its own flag or something

Reply:
because I'm having a problem now that having this would fix. Any time there's been a disparity like this it's miserable to debug. Having this command would help. I've had both circumstances: symlinks without a builtin command, and builtin commands without a symlink.

i could really use this.

Reply:
Can you explain the problem?

Reply:
I have an initramfs with a symlink in /bbin and, for some reason, no corresponding command. This is internal to Google, and something in the build process just plain broke. It would be very convenient to have the ability to have bb tell me the state of its world. 

I did consider the flag but did not want to have bb taking up flag name space. I did consider skipping the symlink but that misses the case of symlink with no compiled-in command -- should never happen, but has, and this is not the first time in the last couple years, and it's maddening to debug. The glob package is included by all kinds of things, hence no additional footprint; the code is pretty tiny here, so I don't see this as a problem.

If you still feel that enumerating symlinks makes no sense, I'll take it out, but ... it directly addresses the problem I'm having now and have had in the past, in a convenient way.

Reply:
fair, ok.

Reply:
Can you make it behind a flag, like:

```
bb --list    # This just lists the commands in bbCmds
bb --verify    # This verifies the symlinks all match up.
bb --fixup    # Creates/removes missing symlinks.
bb        # This displays some help message.
```